spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -173,12 +173,22 @@ protected void sendMessage(Message<?> message) {
 		catch (RuntimeException e) {
 			MessageChannel errorChannel = getErrorChannel();
 			if (errorChannel != null) {
-				this.messagingTemplate.send(errorChannel, new ErrorMessage(e));
+				this.messagingTemplate.send(errorChannel, buildErrorMessage(e));
```

Comment:
Well, that doesn't look so bad (hijacking) because my mind doesn't go somehow so far 😢